.github/workflows/main.yml

```diff
@@ -115,7 +115,7 @@ jobs:
 
       - name: Install Dependencies (Windows)
         if: runner.os == 'Windows'
-        uses: msys2/setup-msys2@cc11e9188b693c2b100158c3322424c4cc1dadea # v2.22.0
+        uses: msys2/setup-msys2@d0e80f58dffbc64f6a3a1f43527d469b4fc7b6c8 # v2.23.0
```

Comment:
Can a comment be added explaining this githashy thing here? Other dependencies don't have such

Reply:
This is used for all GitHub action dependencies. Normally you would do something like
```
 uses: msys2/setup-msys2@v2.23.0
```
To pin it to a specific tag. But tags are mutable so the pinning doesn't really work. We've had cases where a github action deleted a tag and it broke things. And in theory a repo could create a tag, delete it after we upate to it, and them recreate the tag on a later commit and break things without us changing anything.

So we now pin to an actual git hash to avoid this.

We don't do it for other dependencies because their versions aren't mutable, so a version is always guaranteed to be the same. For example, maven jars are immutable and cannot be changed or deleted.